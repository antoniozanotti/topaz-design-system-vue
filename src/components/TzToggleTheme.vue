<script lang="ts" setup>
import TzIcon from "./TzIcon.vue";

export interface TzToggleThemeProps {}

const theme = ref(window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : '');
function handleClick() {
  if (theme.value == 'dark') {
    document.documentElement.classList.remove('dark');
    theme.value = '';
  } else {
    document.documentElement.classList.add('dark');
    theme.value = 'dark';
  }
};

let buttonBackground = "bg-secondary dark:bg-dark-secondary";
let buttonSize =
  "w-[74px] h-[28px] sm:w-[104px] sm:h-[38px] lg:w-[132px] lg:h-[48px]";
let buttonOther = "rounded-full cursor-pointer relative flex justify-around";

let spanAnimation = "transform duration-300";
let spanBackground = "bg-secondary-2 dark:bg-dark-secondary-2";
let spanOther = "block rounded-full absolute";
let spanSize =
  "w-[38px] h-[22px] sm:w-[60px] sm:h-[32px] lg:w-[77px] lg:h-[42px]";
let spanPosition =
  "self-center left-[32px] dark:left-[3px] sm:left-[40px] lg:left-[52px]";
</script>

<template>
  <button @click="handleClick" :class="`${buttonBackground} ${buttonSize} ${buttonOther}`">
    <span
      :class="`${spanAnimation} ${spanBackground} ${spanOther} ${spanSize} ${spanPosition}`"
    />
    <TzIcon
      iconName="MoonIcon"
      variant="primary"
      class="place-self-center"
    />
    <TzIcon
      iconName="SunIcon"
      variant="primary"
      class="place-self-center"
    />
  </button>
</template>
