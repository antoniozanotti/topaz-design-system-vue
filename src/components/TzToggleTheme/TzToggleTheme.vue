<script lang="ts" setup>
import { TzToggleThemeInterface } from "./TzToggleThemeInterface";
import TzIcon from "../TzIcon/TzIcon.vue";

withDefaults(defineProps<TzToggleThemeInterface>(), {});

const buttonBase = "rounded-full cursor-pointer relative flex justify-around";
const buttonBackground = "bg-secondary dark:bg-dark-secondary";
const buttonSize = "w-[104px] h-[38px] lg:w-[132px] lg:h-[48px]";

const spanBase = "block rounded-full absolute";
const spanAnimation = "transform duration-300";
const spanBackground = "bg-secondary-2 dark:bg-dark-secondary-2";
const spanSize = "w-[60px] h-[32px] lg:w-[77px] lg:h-[42px]";
const spanPosition = "self-center left-[40px] dark:left-[3px] lg:left-[52px]";
</script>

<template>
  <button
    :class="`${buttonBase} ${buttonBackground} ${buttonSize}`"
    aria-label="Toggle Theme"
  >
    <span
      :class="`${spanBase} ${spanAnimation} ${spanBackground} ${spanSize} ${spanPosition}`"
    />
    <TzIcon iconName="MoonIcon" variant="primary" class="place-self-center" />
    <TzIcon iconName="SunIcon" variant="primary" class="place-self-center" />
  </button>
</template>
