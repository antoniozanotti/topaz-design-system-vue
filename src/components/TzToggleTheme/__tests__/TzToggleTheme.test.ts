import { describe, it, expect } from "vitest";
import { render, screen } from '@testing-library/vue';
import '@testing-library/jest-dom/vitest';
import TzToggleTheme from '../TzToggleTheme.vue';

describe('TzToggleTheme Component', () => {
  it('should render correctly', () => {
    const { container } = render(TzToggleTheme, { props: { initialTheme: '' } });

    const toggleTheme = screen.getByRole("button");

    expect(toggleTheme).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
