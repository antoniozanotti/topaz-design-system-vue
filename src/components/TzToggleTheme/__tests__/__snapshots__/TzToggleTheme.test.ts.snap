// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TzToggleTheme Component > should render correctly 1`] = `
<div>
  <button
    aria-label="Toggle Theme"
    class="bg-secondary dark:bg-dark-secondary w-[104px] h-[38px] lg:w-[132px] lg:h-[48px] rounded-full cursor-pointer relative flex justify-around"
    initialtheme=""
  >
    <span
      class="transform duration-300 bg-secondary-2 dark:bg-dark-secondary-2 block rounded-full absolute w-[60px] h-[32px] lg:w-[77px] lg:h-[42px] self-center left-[40px] dark:left-[3px] lg:left-[52px]"
    />
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-primary dark:text-dark-primary place-self-center"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.752 15.002A9.718 9.718 0 0118 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 003 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 009.002-5.998z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-primary dark:text-dark-primary place-self-center"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3v2.25m6.364.386l-1.591 1.591M21 12h-2.25m-.386 6.364l-1.591-1.591M12 18.75V21m-4.773-4.227l-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
